backend/src/main/java/hu/flowacademy/konyhatunder/service/RecipeService.java

```diff
@@ -194,6 +194,25 @@ public SearchByIngredientDTO listRecipesByIngredients(List<Ingredient> ingredien
         return sortRecipesByName(response);
     }
 
+    public void recommendARecipe(String recommend, String id) {
+        Recipe recipe = recipeRepository.findById(id).orElseThrow(() -> new ValidationException("Nincs ilyen ID-val rendelkező recept!"));
```

Comment:
Ezt az egészet tranzakcióba kellene rakni, mert ha egyszerre két felhasználó hívja, akkor az egyik ajánlása elveszhet.

1. A meghívja, kiolvassa, hogy 0 ajánlás van.
2. B meghívja, kiolvassa, hogy 0 ajánlás van.
3. A megnöveli eggyel és beírja => 0+1 = 1 ajánlás van.
4. B megnöveli eggyel és beírja => 0+1 = 1 ajánlás van.

Ha az egyik nem növel, hanem csökkent, akkor még rosszabb a helyzet, 2 is lehet az eredmény a kívánt és tényleges között a különbség.

Reply:
Es a recommend lehetne boolean, egyszerusodne a kod. 
Meg itt annyit megjegyznek hogy ha nem akartok negativ erteket megenegedni akkor eleg 0-ra settelni az erteket es nem hozzanyulni ha lejjebb vinnek, de null-ra direkt ne setteljuk, csak plusz hibalehetosegeket hoz be. 
Ha egy "user"nek nem akartok tobb vote-ot engedni akkor pedig jo ideiglenes hack lehet, hogy locale storage-ben letaroljatok az id-t egy true/false-al igy checkolni tudtok hogy az adott bongeszobol merre valtoztattak egy receptet. A demo-ra ez nem egy rossz megoldas.